### ISSUE TITLE
unexpected logical expression result

### ISSUE THREAD

**Issue by vjeantet** [Knetic/govaluate#8]
Hello
 || and && operators does not seems to work as expected 

```go
    expression, _ := govaluate.NewEvaluableExpression("foo == true || bar == true")
	parameters := make(map[string]interface{}, 8)
	parameters["foo"] = true
	parameters["bar"] = false
	result, _ := expression.Evaluate(parameters)
	log.Printf("actual %t, expected : true \n", result)

	expression, _ = govaluate.NewEvaluableExpression("foo > 10  && bar > 10")
	parameters = make(map[string]interface{}, 8)
	parameters["foo"] = 1
	parameters["bar"] = 11
	result, _ = expression.Evaluate(parameters)
	log.Printf("actual %t, expected : false \n", result)
```

this code will evaluate `false` for the first result, and `true` for the second one.

**Comment by Knetic** [comment]
Thanks for the report, especially the sample! There was a gap in my tests, and it made me think I could be clever and get by without doing a type assertion when evaluating logical operators. For whatever reason, nobody's hit this case until today.

Ought to be fixed as of dc6cef2046fa304157beca38bcc8635e63e183e1, I added test cases for your examples and everything ought to work as expected. Let me know if you have any problems with the master branch.

**Comment by vjeantet** [comment]
It works !  Thanks for being so quick to fix it !

Nice piece of code !
